packages/client/src/containers/RelatedItems/RelatedItems.Container.js

```diff
@@ -1,33 +1,77 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
 import PropTypes from 'prop-types';
 import ProductCard from '../../components/ProductCard/ProductCard.component';
 import { apiURL } from '../../apiURL';
+import { ViewBackCarousel } from '../../components/ViewCarousel/ViewBackCarousel/ViewBackCarousel';
+import { ViewAfterCarousel } from '../../components/ViewCarousel/ViewForwadCarousel/ViewAfterCarousel';
 import './RelatedItems.styles.css';
 
 export const RelatedItems = ({ category }) => {
   const [categoryProducts, setCategoryProducts] = useState([]);
+  const [offSet, setOffset] = useState(0);
+  const [hasMoreProducts, setHasMoreProducts] = useState(true);
 
   useEffect(() => {
-    const fetchProductsByCategory = async () => {
-      const response = await fetch(`${apiURL()}/products?category=${category}`);
+    async function fetchProductsByCategory() {
+      const response = await fetch(
+        `${apiURL()}/products?category=${category}&limit=3&offset=${offSet}`,
+      );
       const categoryProductData = await response.json();
+      if (categoryProductData.length === 0) {
+        if (categoryProductData.length === 0 && offSet === 0) {
+          await fetchAllProducts();
+        }
+        setHasMoreProducts(false);
+      } else if (categoryProductData.length < 3) {
+        setHasMoreProducts(false);
+        setCategoryProducts(categoryProductData);
+      } else {
+        setHasMoreProducts(true);
+        setCategoryProducts(categoryProductData);
+      }
+    }
+    async function fetchAllProducts() {
+      const response = await fetch(`${apiURL()}/products`);
```

Comment:
```suggestion
      const response = await fetch(`${apiURL()}/products?limit=3`);
```

If I look at the URL above (`api/products?category=${category}&limit=3&offset=${offSet}`) then I would assume that I could send limit here as well rather than using `slice`.
If that's not possible, we should change it so that it **is** possible because query parameters should not vary like that.

Reply:
@miltonhultgren yes in the endpoint for getAllproducts we don't have any limit query parameter so I haven't used there and used slice

Reply:
We will update it when the endpoint is updated but for now for this PR I think to do. with slice